devel/README.md

```diff
@@ -1,7 +1,341 @@
+The push is allowed to proceed only if the hook succeeds.
+
+### Create a branch and make changes
+
+```shell
+git checkout -b my-feature
+# Make your code changes
+```
+
+### Keeping your fork in sync
+
+```shell
+git fetch upstream
+git rebase upstream/master
+```
+
+Note: If you have write access to the main repositories 
+(eg github.com/istio/mixer), you should modify your git configuration so
+that you can't accidentally push to upstream:
+
+```shell
+git remote set-url --push upstream no_push
+```
+
+### Committing changes to your fork
+
+When you're happy with some changes, you can commit them to your repo:
+
+```shell
+git add .
+git commit
+```
+Then push the change to the fork. When prompted for authentication, using your
+github username as usual but the personal access token as your password if you
+have not setup ssh keys. Please
+follow [these
+instructions](https://help.github.com/articles/caching-your-github-password-in-git/#platform-linux)
+if you want to cache the token.
+
+```shell
+git push -f origin my-feature
+```
+
+### Creating a pull request
+
+1. Visit https://github.com/$GITHUB_USER/$COMPONENT
+2. Click the "Compare & pull request" button next to your "my-feature" branch.
+
+### Getting a code review
+
+Once your pull request has been opened it will be assigned to one or more
+reviewers. Those reviewers will do a thorough code review, looking for
+correctness, bugs, opportunities for improvement, documentation and comments,
+and style.
+
+Very small PRs are easy to review.  Very large PRs are very difficult to
+review. GitHub has a built-in code review tool, which is what most people use.
+
+### When to retain commits and when to squash
+
+Upon merge, all git commits should represent meaningful milestones or units of
+work. Use commits to add clarity to the development and review process.
+
+Before merging a PR, squash any "fix review feedback", "typo", and "rebased"
+sorts of commits. It is not imperative that every commit in a PR compile and
+pass tests independently, but it is worth striving for. For mass automated
+fixups (e.g. automated doc formatting), use one or more commits for the
+changes to tooling and a final commit to apply the fixup en masse. This makes
+reviews much easier.
+
+## Using the code base
+
+### Building the code
+
+Using Mixer as an example:
+
+```shell
+cd $(ISTIO)/mixer
+bazel build ...
+```
+
+This figures out what it needs to do and does not need any input from you.
+
+### Setup bazel and go links
+
+Symlinks bazel artifacts into the standard go structure so standard go
+tooling functions correctly
+
+```shell
+./bin/bazel_to_go.py
+```
+(You can safely ignore some errors like
+`com_github_opencontainers_go_digest Does not exist`)
+
+### Cleaning outputs
+
+You can delete any build artifacts with:
+
+```shell
+bazel clean
+```
+### Running tests
+
+You can run all the available tests with:
+
+```shell
+bazel test ...
+```
+### Getting coverage numbers
```

Comment:
Does this apply to all repos? Same question of the other stuff like "make fmt" and friends. Those are setup for Mixer, and unless they've been copied elsewhere, other repos aren't on that plan (yet).